code/game/turfs/simulated/floor.dm

```diff
@@ -46,7 +46,8 @@ var/list/wood_icons = list("wood","wood-broken")
 		builtin_tile = new floor_tile
 
 	spawn(5)
-		update_visuals()
+		if(istype(src, /turf/simulated)) //you are probably looking at this and thinking "what the fuck", because that's the appropriate response
+			update_visuals()             //however, shityond means that turfs can change src and spawned procs will try to run on it, and only sim turfs have update_visuals()
 

```

Comment:
If this was java, and a situation was found where `this` was suddenly pointing to a different object of a different type, it would be front page news on the tech sites.